pkgs/applications/science/biology/kssd/default.nix

```diff
@@ -1,38 +1,46 @@
-{ lib, stdenv, fetchFromGitHub, fetchpatch, zlib, automake, autoconf, libtool }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, zlib
+, kssd
+, runCommand
+}:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "kssd";
-  version = "1.1";
+  version = "2.21";
 
   src = fetchFromGitHub {
     owner = "yhg926";
     repo = "public_kssd";
-    rev = "v${version}";
-    sha256 = "sha256-8jzYqo9LXF66pQ1EIusm+gba2VbTYpJz2K3NVlA3QxY=";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-D/s1jL2oKE0rSdRMVljskYFsw5UPOv1L95Of+K+e17w=";
   };
 
-  patches = [
-    # Pull upstream patch for -fno-common toolchain support:
-    #   https://github.com/yhg926/public_kssd/pull/9
-    (fetchpatch {
-      name = "fno-common.patch";
-      url = "https://github.com/yhg926/public_kssd/commit/cdd1e8aae256146f5913a3b4c723b638d53bdf27.patch";
-      sha256 = "sha256-HhaTRqPfKR+ouh0PwEH6u22pbuqbX2OypRzw8BXm0W4=";
-    })
-  ];
-
-  nativeBuildInputs = [ autoconf automake ];
-  buildInputs = [ zlib libtool ];
+  buildInputs = [ zlib ];
 
   installPhase = ''
-      install -vD kssd $out/bin/kssd
+    runHook preInstall
+
+    install -vD kssd $out/bin/kssd
+
+    runHook postInstall
   '';
 
+  passthru.tests = {
+    simple = runCommand "${finalAttrs.pname}-test" { } ''
```

Comment:
ok, i can reproduce segfault locally

```
/build/.attr-0l2nkwhif96f51f4amnlf414lhl4rv9vh8iffyp431v6s28gsr90: line 2:     6 Aborted                 (core dumped) /nix/store/a56iln8xbjz38ixfjz2w804y0mpix4kd-kssd-2.21/bin/kssd dist -L /nix/store/97nwzkrxz4ir9rp6b1nrar5mrnp3l0ha-source/shuf_file/L3K10.shuf -r /nix/store/97nwzkrxz4ir9rp6b1nrar5mrnp3l0ha-source/test_fna/seqs1 -o $out/reference
error: builder for '/nix/store/ywfwn7ix85h6xq6v44x83i25mm312x9r-kssd-test.drv' failed with exit code 134;
       last 2 log lines:
       > *** buffer overflow detected ***: terminated
       > /build/.attr-0l2nkwhif96f51f4amnlf414lhl4rv9vh8iffyp431v6s28gsr90: line 2:     6 Aborted                 (core dumped) /nix/store/a56iln8xbjz38ixfjz2w804y0mpix4kd-kssd-2.21/bin/kssd dist -L /nix/store/97nwzkrxz4ir9rp6b1nrar5mrnp3l0ha-source/shuf_file/L3K10.shuf -r /nix/store/97nwzkrxz4ir9rp6b1nrar5mrnp3l0ha-source/test_fna/seqs1 -o $out/reference
       For full logs, run 'nix log /nix/store/ywfwn7ix85h6xq6v44x83i25mm312x9r-kssd-test.drv'.

```

Reply:
I have tried changing the parameters and truncating the test data, but I cannot reduce the memory usage below 2 GB.
CI is grayed out, but would such a test be acceptable?

Reply:
this fixes segfault https://github.com/yhg926/public_kssd/pull/11

Reply:
Awesome! Thank you!